src/modules/Experiences/ExperiencesList/helpers.js

```diff
@@ -84,4 +84,61 @@ const configureForm = (expType) => {
   }
 };
 
-export { formatMonthYearDate, getWorkLength, configureForm };
+const monthScore = {
+  January: 1,
+  February: 2,
+  March: 3,
+  April: 4,
+  May: 5,
+  June: 6,
+  July: 7,
+  August: 8,
+  September: 9,
+  October: 10,
+  November: 11,
+  December: 12,
+};
+
+const getMonthScore = (experiences) => {
+  experiences.map((exp) => {
+    exp.start_month_score = monthScore[exp.start_month];
+
+    // exclude end_month === 'none'
+    if (exp.is_current === false) {
+      exp.end_month_score = monthScore[exp.end_month];
+    }
+  });
+
+  return experiences;
+};
+
+const sortExperiences = (experiences) => {
+  const experiencesWithScores = getMonthScore(experiences);
+
+  const sortedExperiences = experiencesWithScores.sort(function(exp1, exp2) {
+    // if is_current is True or have the same end month/year
+    // sort from start month/year (newest experience)
+    if (exp1.end_month === exp2.end_month && exp1.end_year === exp2.end_year) {
+      if (exp1.start_year > exp2.start_year) return -1;
+      if (exp1.start_year < exp2.start_year) return 1;
+      if (exp1.start_month_score > exp2.start_month_score) return -1;
+      if (exp1.start_month_score < exp2.start_month_score) return 1;
+    } else if (exp1.is_current === false && exp2.is_current === false) {
+      if (exp1.end_year > exp2.end_year) return -1;
+      if (exp1.end_year < exp2.end_year) return 1;
+      if (exp1.end_month_score > exp2.end_month_score) return -1;
+      if (exp1.end_month_score < exp2.end_month_score) return 1;
+    }
+    if (exp1.is_current === true) return -1;
```

Comment:
This code is really hard to follow. Some tactical suggestions:

Specifically when writing a sort key function, it's easiest if the logic follows a consistent pattern. I think here you have a pretty simple pattern, called a `lexicographic ordering` (basically the way you alphabetize words). It means you first compare the first thing, and sort by that, then if the first thing is equal you move onto the second thing.

Based on your test, it looks like the ordering you want is this:

1. is_current (true goes above false)
2. end_year
3. end_month
4. start_year
5. start_month

You can then write some predictable sort code by using the pattern:
```javascript
if (exp1.is_current !== exp2.is_current) {
 // return ...
} else if (exp1.end_year !== exp2.start_year) {
 // etc...
```

Reply:
Done.